### PR TITLE
Use __future__ annotations in dataset

### PR DESCRIPTION
**Pull request by jenshnielsen**

### CODE REVIEW

File: qcodes/dataset/data_export.py [hunk 1]
```diff
@@ -20,12 +23,13 @@ class DSPlotData(TypedDict):
...
     data: np.ndarray
-    shape: Optional[Tuple[int, ...]]
+    shape: tuple[int, ...] | None
 
 
 @deprecate(alternative="ndarray.flatten()")
-def flatten_1D_data_for_plot(rawdata: Union[Sequence[Sequence[Any]],
-                                            np.ndarray]) -> np.ndarray:
+def flatten_1D_data_for_plot(
+    rawdata: (Sequence[Sequence[Any]] | np.ndarray),
```
Comment (astafan8) [inline qcodes/dataset/data_export.py:31]:
```suggestion
    rawdata: Sequence[Sequence[Any]] | np.ndarray,
```

File: qcodes/dataset/plotting.py [hunk 2]
```diff
@@ -94,13 +95,13 @@ def heatmaphandler(**kwargs: Any) -> Any:
...
 def plot_dataset(
     dataset: DataSetProtocol,
-    axes: Optional[Union[matplotlib.axes.Axes, Sequence[matplotlib.axes.Axes]]] = None,
-    colorbars: Optional[
-        Union[matplotlib.colorbar.Colorbar, Sequence[matplotlib.colorbar.Colorbar]]
-    ] = None,
+    axes: matplotlib.axes.Axes | Sequence[matplotlib.axes.Axes] | None = None,
+    colorbars: matplotlib.colorbar.Colorbar
+    | Sequence[matplotlib.colorbar.Colorbar]
+    | None = None,
```
Comment (astafan8) [inline qcodes/dataset/plotting.py:101]: (no action needed) not sure what to think about this formatting, i think that i preferred what black does with long experessions inside brackets.

**Review by astafan8 — APPROVED** [review]

File: qcodes/dataset/data_export.py (hunk 1 above)
Comment (jenshnielsen) [reply qcodes/dataset/data_export.py]: Done

File: qcodes/dataset/plotting.py (hunk 2 above)
Comment (jenshnielsen) [reply qcodes/dataset/plotting.py]: Replaced by short form to avoid these overly long lines

**Pull request merged**